rocksdb_admin/detail/file_watcher_manager.h

```diff
@@ -0,0 +1,52 @@
+/// Copyright 2019 Pinterest Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+/// http://www.apache.org/licenses/LICENSE-2.0
+
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+
+//
+// @author rajathprasad (rajathprasad@pinterest.com)
+//
+
+#include <memory>
+#include <mutex>
+#include <unordered_map>
+
+namespace kafka {
+  class KafkaBrokerFileWatcher;
+}
+
+namespace admin {
+namespace detail {
+
+  // FileWatcherManager is a singleton class which manages the kafka broker
+  // file watchers for this process.
+  class FileWatcherManager {
```

Comment:
nit: Can we rename it to something like KafkaBrokerFileWatcher?

Reply:
Renamed to KafkaBrokerFileWatcherManager. Initially had decided against it since its a mouthful.